### ISSUE TITLE
bench-command.sh refuses every self-supervising Python measurement entry point: the wrapper never passes --entrypoint

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#1688]
> _Issue authored by Claude (Anthropic agent) on behalf of @ohdearquant._

`scripts/bench-command.sh` refuses, rather than measures, whenever the command it wraps is one of the Python measurement entry points that supervise themselves. The refusal names an environment variable, so the composition that caused it is not visible in the message.

This is the sibling of #1643 / #1563 and not a duplicate of either: those are about the Rust `gpu_test_lock()` waiting on a lock its own supervisor holds. This one never reaches a lock. It is the Python entry-point handoff, and it fails in the opposite direction — fast and loud rather than silent for thirty minutes.

## Reproduction

```
$ PYTHON_BIN=<python3.13> scripts/bench-command.sh --label probe -- \
    <python3.13> scripts/bench_linkedin_post.py --help
bench-supervision: LATTICE_BENCH_SUPERVISOR_FD is not set; refusing to measure
$ echo $?
2
```

Run on `8054a3cd9208757d84b6454fb56f0d838e5e849b`.

## Mechanism

`scripts/bench-command.sh:48,50` are the two `exec` branches, ordinary and `--durable`. Neither passes `--entrypoint`:

```
exec "$PYTHON_BIN" "$HELPER" run --label "$LABEL" --quiet ${GPU_HANDOFF[@]+"${GPU_HANDOFF[@]}"} -- "$@"
exec "$PYTHON_BIN" "$HELPER" run --label "$LABEL"         ${GPU_HANDOFF[@]+"${GPU_HANDOFF[@]}"} -- "$@"
```

The flag exists and is passed elsewhere — `scripts/lib/bench-supervision.sh:32,34` and `scripts/bench-compare.sh:64` — which is what makes the absence here a gap rather than a design.

Without it, `run_supervised` exports `LATTICE_BENCH_LOCK_STATUS` to the child and does not export `LATTICE_BENCH_SUPERVISOR_FD`. That combination is observable directly:

```
$ scripts/bench-command.sh --label probe -- /usr/bin/env | grep LATTICE_BENCH
LATTICE_BENCH_LOCK_STATUS=.../.cache/bench-supervision/probe.status
```

One variable, not three. So in `ensure_python_entrypoint` (`scripts/lib/bench_supervision.py:480`) the `STATUS_ENV not in os.environ` test is false — the child correctly concludes it is already supervised and does not re-exec — and it then calls `sample_measurement_handoff` (`:229`), which reads the receipt, validates the lock paths, and calls `_sample_open_pipe_writer`. That raises at `:186` because `LATTICE_BENCH_SUPERVISOR_FD` is unset, and the `except SupervisionError` at `:491` prints the message above and exits with the refusal code.

The later "existing handoff is lock-only" diagnostic at `:494` is unreachable from this composition: it requires a valid handoff with insufficient quiet-policy assurance, and this path has no handoff at all.

The general shape is that the child can see *that* a supervisor exists but cannot receive the particular authority this invocation needs, and the two conditions are not distinguished anywhere. `FDS_ENV` is also absent here but never consulted — `sample_measurement_handoff` does not check it, so the descriptor-inheritance branch in `verify_supervision` (`:177`) is not the one that fires.

## Affected population

Twelve of the thirty-four Python scripts under `scripts/` call `ensure_python_entrypoint`, and they are exactly the set this composition refuses:

```
$ git grep -l -w ensure_python_entrypoint -- scripts | grep -v '^scripts/lib/' | wc -l
12
```

`bench_cpu_flagship_supervisor.py`, `bench_decode_adapters_agentic.py`, `bench_decode_adapters_apples_precise.py`, `bench_decode_adapters_apples_to_apples.py`, `bench_decode_adapters_context_scaling.py`, `bench_decode_adapters_q4_apples.py`, `bench_decode_harness.py`, `bench_linkedin_post.py`, `compare_logits.py`, `e2e_parity_check.py`, `fake_quant_pilot.py`, `perf_governor.py`. The other twenty-two do not call it, so the predicate discriminates.

The shell entry points (`bench-ci.sh`, `bench-gate.sh`, `bench_decode_slopefit.sh`, `bench_quality.sh`, `e2e-parity-local.sh`) are not in this set. They reach the same helper through `bench_supervise_entry`, which does pass `--entrypoint` — they are the working form, and the contrast is the evidence that the wrapper's omission is the defect rather than the design.

## Reachability, stated honestly

No committed script composes the wrapper around one of these; every caller found is prose. `CLAUDE.md` documents `scripts/bench-command.sh --label <name> -- <command>` as the general form for an ad-hoc measurement command, and `scripts/perf_governor.README.md:51` says outright that `bench-command.sh` is redundant for that script — which is documentation standing in for a guard. So this is a latent composition hazard reachable by a reader following the general instruction, not a live break in an automated path.

That it fails closed is the good half. The cost is diagnostic: the message names a variable the reader did not set and cannot be expected to know about, on a command the top-level guidance presents as the generic one.

## Suggested direction

Either of these would close it, and they are not equivalent:

1. Have `bench-command.sh` detect that its command is a self-supervising entry point and forward `--entrypoint`, so the composition works rather than refuses. This is the larger change and needs a rule for what counts as such a command.
2. Keep the refusal and make it name the composition: when `STATUS_ENV` is present and `SUPERVISOR_FD_ENV` is not, say that the entry point is already under a supervisor that did not pass an entry-point handoff, and name the two ways out (drop the wrapper, or use `scripts/lib/bench-supervision.sh`). Cheap, and it turns a variable name into an instruction.

If (2) is chosen, `scripts/bench-measurements.toml` should record that the wrapper is not a valid outer layer for that class of target, so the classification and the message agree.
